src/main/scala/nl.knaw.dans.easy.solr4files/package.scala

```diff
@@ -176,7 +179,9 @@ package object solr4files extends DebugEnhancedLogging {
       if (left.getProtocol.toLowerCase == "file")
         Try(new File(left.getPath).length)
       else {
-        val fileSizesUrl = left.toString.replace("bags", "bags/filesizes")
+        val urlPattern = "^.*/stores/.*/bags/.*$".r
+        val itemUrl = urlPattern findFirstIn(left.toString) getOrElse(Failure(InvalidItemUrlException(left)))
+        val fileSizesUrl = itemUrl.toString.replaceFirst("/bags/", "/bags/filesizes/")
```

Comment:
This way we can see what is the expected pattern of the url and how the new url is created from it.  But I am not sure if this is needed to make it clear for the reader.  We could also simply add a comment:) What do you think?

Reply:
In my experience comments should be avoided if possible. The [Clean Code book](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882) describes the problem with comments in a very nice and recognizable way. One of the code smells is called "lying comment" if a remember it correctly, i.e. a comment that was at some point true but was not kept in sync with the code it is supposed to describe. Another type of comment that I encounter all too often is what you could call "obscure comment": something that might have had some meaning to the author, but is completely mystifying to anybody else.

By choosing variable and function names that describe what you are doing, comments are usually easily avoided.

Reply:
Yep, that is also so that comments easily start to stink...
But.. do you agree with the code change?